@navikt/core/react/src/layout/base/BasePrimitive.tsx

```diff
@@ -0,0 +1,197 @@
+import cl from "clsx";
+import React from "react";
+import { Slot } from "../../util/Slot";
+import { getResponsiveProps, getResponsiveValue } from "../utilities/css";
+import { ResponsiveProp, SpacingScale } from "../utilities/types";
+
+export interface BasePrimitiveProps {
+  children: React.ReactElement;
+  className?: string;
+  /** Padding around children. Accepts a spacing token or an object of spacing tokens for different breakpoints.
+   * @example
+   * padding='4'
+   * padding={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
+   */
+  padding?: ResponsiveProp<SpacingScale>;
+  /** Horizontal padding around children. Accepts a spacing token or an object of spacing tokens for different breakpoints.
+   * @example
+   * paddingInline='4'
+   * paddingInline='4 5'
+   * paddingInline={{xs: '0 32', sm: '3', md: '4 5', lg: '5', xl: '6'}}
+   */
+  paddingInline?: ResponsiveProp<
+    SpacingScale | `${SpacingScale} ${SpacingScale}`
+  >;
+  /** Vertical padding around children. Accepts a spacing token or an object of spacing tokens for different breakpoints.
+   * @example
+   * paddingBlock='4'
+   * paddingBlock='4 5'
+   * paddingBlock={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
+   */
+  paddingBlock?: ResponsiveProp<
+    SpacingScale | `${SpacingScale} ${SpacingScale}`
+  >;
+  /**
+   * CSS `width`
+   */
+  width?: ResponsiveProp<string>;
+  /**
+   * CSS `min-width`
+   */
+  minWidth?: ResponsiveProp<string>;
+  /**
+   * CSS `max-width`
+   */
+  maxWidth?: ResponsiveProp<string>;
+  /**
+   * CSS `height`
+   */
+  height?: ResponsiveProp<string>;
+  /**
+   * CSS `min-height`
+   */
+  minHeight?: ResponsiveProp<string>;
+  /**
+   * CSS `max-height`
+   */
+  maxHeight?: ResponsiveProp<string>;
+  /**
+   * CSS `position`
+   */
+  position?: ResponsiveProp<
+    "static" | "relative" | "absolute" | "fixed" | "sticky"
+  >;
+  /**
+   * CSS `inset`
+   */
+  inset?: ResponsiveProp<SpacingScale>;
+  /**
+   * CSS `top`
+   */
+  top?: ResponsiveProp<SpacingScale>;
+  /**
+   * CSS `right`
+   */
+  right?: ResponsiveProp<SpacingScale>;
+  /**
+   * CSS `bottom`
+   */
+  bottom?: ResponsiveProp<SpacingScale>;
+  /**
+   * CSS `left`
+   */
+  left?: ResponsiveProp<SpacingScale>;
```

Comment:
Suggest adding JSDOC in same style as padding, and accept two values in `inset`: ```ResponsiveProp<SpacingScale | `${SpacingScale} ${SpacingScale}`>```:

```suggestion
  /**
   * CSS `inset`.
   * Accepts a [spacing token](https://aksel.nav.no/grunnleggende/styling/design-tokens#0cc9fb32f213)
   * or an object of spacing tokens for different breakpoints.
   * @example
   * inset='4'
   * inset='4 5'
   * inset={{xs: '0 32', sm: '3', md: '4 5', lg: '5', xl: '6'}}
   */
  inset?: ResponsiveProp<SpacingScale | `${SpacingScale} ${SpacingScale}`>;
  /**
   * CSS `top`
   * Accepts a [spacing token](https://aksel.nav.no/grunnleggende/styling/design-tokens#0cc9fb32f213)
   * or an object of spacing tokens for different breakpoints.
   * @example
   * top='4'
   * top={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
   */
  top?: ResponsiveProp<SpacingScale>;
  /**
   * CSS `right`
   * Accepts a [spacing token](https://aksel.nav.no/grunnleggende/styling/design-tokens#0cc9fb32f213)
   * or an object of spacing tokens for different breakpoints.
   * @example
   * right='4'
   * right={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
   */
  right?: ResponsiveProp<SpacingScale>;
  /**
   * CSS `bottom`
   * Accepts a [spacing token](https://aksel.nav.no/grunnleggende/styling/design-tokens#0cc9fb32f213)
   * or an object of spacing tokens for different breakpoints.
   * @example
   * bottom='4'
   * bottom={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
   */
  bottom?: ResponsiveProp<SpacingScale>;
  /**
   * CSS `left`
   * Accepts a [spacing token](https://aksel.nav.no/grunnleggende/styling/design-tokens#0cc9fb32f213)
   * or an object of spacing tokens for different breakpoints.
   * @example
   * left='4'
   * left={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
   */
  left?: ResponsiveProp<SpacingScale>;
```